src/lib/index/wb_cache.cpp

```diff
@@ -418,52 +433,79 @@ void IndexWBCache::recover(sisl::byte_view sb) {
     // This has to be done before doing any repair, because repair can allocate blkids and we don't want to allocate
     // the same blkid which could clash with the blkid next in the buf list.
     //
-    // On the second pass, we only take the new nodes/bufs and then repair their up buffers, if needed.
-    std::vector< IndexBufferPtr > l0_bufs;
+    // On the second pass, we only take part of the parents/siblings and then repair them, if needed.
+    std::vector<IndexBufferPtr> pending_bufs;
+    std::vector<IndexBufferPtr> deleted_bufs;
     for (auto const& [_, buf] : bufs) {
-        if (buf->m_node_freed || (buf->m_created_cp_id == icp_ctx->id())) {
+        if (buf->m_node_freed) {
+            // Freed node
+            if (buf->m_bytes == nullptr) {
+                buf->m_bytes = hs_utils::iobuf_alloc(m_node_size, sisl::buftag::btree_node, m_vdev->align_size());
+                m_vdev->sync_read(r_cast<char *>(buf->m_bytes), m_node_size, buf->blkid());
+            }
```

Comment:
if you read the buf->m_bytes , them the is valid and modified cp will not be set in was_commited and you need to check it later (which is not NIT). Also you need to read sync_read twice which is unnecessary and cause performance degradation